src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -124,6 +127,13 @@ bool shared_node_optimization(const shared_ptr<Model>& model) {
                 auto& shared_nodes = item.second;
```

Comment:
I have just remembered that there is an issue in SymbolicOptimization in cases when the target inputs are the same node.
e.g. Split -> Concat 
probably also in Split -> Add (Split_out_1, Split_out_2)

maybe it's better to use Inputs instead of Nodes to compare, or set of Nodes

Reply:
Well, it seems that this is a separate issue and should be investigated and fixed separately. I didn't not make it worse here.

Reply:
yes, I will create a ticket